test/libyul/DominatorTest.cpp

```diff
@@ -0,0 +1,520 @@
+/*
+    This file is part of solidity.
+    solidity is free software: you can redistribute it and/or modify
+    it under the terms of the GNU General Public License as published by
+    the Free Software Foundation, either version 3 of the License, or
+    (at your option) any later version.
+    solidity is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
+    You should have received a copy of the GNU General Public License
+    along with solidity.  If not, see <http://www.gnu.org/licenses/>.
+*/
+/**
+ * Unit tests for the algorithm to find dominators from a graph.
+ */
+#include <libyul/backends/evm/Dominator.h>
+
+#include <test/libsolidity/util/SoltestErrors.h>
+
+#include <boost/test/unit_test.hpp>
+
+#include <range/v3/range/conversion.hpp>
+#include <range/v3/view/transform.hpp>
+
+using namespace solidity::yul;
+
+namespace solidity::yul::test
+{
+
+struct ImmediateDominatorTest
+{
+	struct Vertex {
+		std::string name;
+		std::vector<Vertex*> successors;
+
+		bool operator<(Vertex const& _other) const
+		{
+			return name < _other.name;
+		}
+	};
+
+	typedef std::pair<std::string, std::string> Edge;
+
+	struct ForEachVertexSuccessorTest {
+		template<typename Callable>
+		void operator()(Vertex _v, Callable&& _callable) const
+		{
+			for (auto const& w: _v.successors)
+				_callable(*w);
+		}
+	};
+
+	size_t numVertices;
+	Vertex* entry;
```

Comment:
Please remember to always initialize variables of primitive data types:
```suggestion
	size_t numVertices = 0;
	Vertex* entry = nullptr;
```

or
```suggestion
	size_t numVertices{};
	Vertex* entry{};
```